src/ResultDecoder.php

```diff
@@ -135,31 +139,39 @@ public function extractCrumb(string $responseBody): string
         throw new ApiException('Could not extract crumb from response', ApiException::MISSING_CRUMB);
     }
 
-    public function transformHistoricalDataResult(string $responseBody): array
+    private function validateHeaderLines(string $responseBody, array $expectedHeader): array
     {
         $lines = array_map('trim', explode("\n", trim($responseBody)));
         $headerLine = array_shift($lines);
-        $expectedHeaderLine = implode(',', self::HISTORICAL_DATA_HEADER_LINE);
+        $expectedHeaderLine = implode(',', $expectedHeader);
         if ($headerLine !== $expectedHeaderLine) {
             throw new ApiException(sprintf('CSV header line did not match expected header line, given: %s, expected: %s', $headerLine, $expectedHeaderLine), ApiException::INVALID_RESPONSE);
         }
 
-        return array_map(function ($line) {
-            return $this->createHistoricalData(explode(',', $line));
-        }, $lines);
+        return $lines;
     }
 
-    private function createHistoricalData(array $columns): HistoricalData
+    private function validateDate(array $columns): \DateTime
     {
-        if (7 !== \count($columns)) {
-            throw new ApiException('CSV did not contain correct number of columns', ApiException::INVALID_RESPONSE);
-        }
-
         try {
-            $date = new \DateTime($columns[0], new \DateTimeZone('UTC'));
+            return new \DateTime($columns[0], new \DateTimeZone('UTC'));
         } catch (\Exception $e) {
             throw new ApiException(sprintf('Not a date in column "Date":%s', $columns[0]), ApiException::INVALID_VALUE);
         }
+    }
```

Comment:
I don't like so much how this is bound to `$column[0]`. I'd say let's change this to `validateDate(string $value): \DateTime` instead and let the calling method decide what kind of value to pass.